R/aaa_models.R

```diff
@@ -134,25 +135,114 @@ check_mode_val <- function(mode) {
 }
 
 
-stop_incompatible_mode <- function(spec_modes) {
+stop_incompatible_mode <- function(spec_modes, eng) {
+  if (is.null(eng)) {
+    msg <- glue::glue(
+      "Available modes are: ",
+      glue::glue_collapse(glue::glue("'{spec_modes}'"), sep = ", ")
+    )
+  } else {
+    msg <- glue::glue(
+      "Available modes for engine {eng} are: ",
+      glue::glue_collapse(glue::glue("'{spec_modes}'"), sep = ", ")
+    )
+  }
+
+  rlang::abort(msg)
+}
```

Comment:
When we use it with the model class passed as the `eng` argument, it prints messages like "Available modes for *engine linear_reg* are:"  so here's a suggestion on how to change that:
```suggestion
stop_incompatible_mode <- function(spec_modes, eng = NULL, cls = NULL) {
  if (is.null(eng) & is.null(cls)) {
    msg <- "Available modes are: "
  }
  if (!is.null(eng) & is.null(cls)) {
    msg <- glue::glue("Available modes for engine {eng} are: ")
  }
  if (is.null(eng) & !is.null(cls)) {
    msg <- glue::glue("Available modes for model type {cls} are: ")
  }
  if (!is.null(eng) & !is.null(cls)) {
    msg <- glue::glue("Available modes for model type {cls} with engine {eng} are: ")
  }

  msg <- glue::glue(
    msg,
    glue::glue_collapse(glue::glue("'{spec_modes}'"), sep = ", ")
  )
  rlang::abort(msg)
}
```